src/Core/Picking.js

```diff
@@ -0,0 +1,98 @@
+import * as THREE from 'three';
+import TileMesh from './TileMesh';
+import RendererConstant from '../Renderer/RendererConstant';
+import { unpack1K } from '../Renderer/LayeredMaterial';
+
+// TileMesh picking support function
+function screenCoordsToNodeId(view, tileLayer, mouse) {
+    const dim = view.mainLoop.gfxEngine.getWindowSize();
+
+    mouse = mouse || new THREE.Vector2(Math.floor(dim.x / 2), Math.floor(dim.y / 2));
+
+    const restore = tileLayer.level0Nodes.map(n => n.pushRenderState(RendererConstant.ID));
+
+    // Prepare state
+    const prev = view.camera.camera3D.layers.mask;
+    view.camera.camera3D.layers.mask = 1 << tileLayer.threejsLayer;
+
+    var buffer = view.mainLoop.gfxEngine.renderViewTobuffer(
+        view,
+        view.mainLoop.gfxEngine.fullSizeRenderTarget,
+        mouse.x, dim.y - mouse.y,
+        1, 1);
+
+    restore.forEach(r => r());
+
+    view.camera.camera3D.layers.mask = prev;
+
+    var depthRGBA = new THREE.Vector4().fromArray(buffer).divideScalar(255.0);
+
+    // unpack RGBA to float
+    var unpack = unpack1K(depthRGBA, Math.pow(256, 3));
+
+    return Math.round(unpack);
+}
+
+export default {
+    pickTilesAt: (_view, mouse, layer) => {
```

Comment:
The picking function shouldn't have as parameter the layer but the objects to pick.
Because this function must be usable without the notion of layer